testsuite/MDAnalysisTests/analysis/test_distances.py

```diff
@@ -106,43 +108,64 @@ def test_box_sparse(self, coord, box, shape, res_pbc):
         assert_equal(contacts.toarray(), res_pbc)
 
 
+
 class TestDist(object):
 
     @staticmethod
     @pytest.fixture()
     def ag():
         u = MDAnalysis.Universe(GRO)
-        return u.atoms[:20]
+        return u.atoms[1:10]
 
     # TODO: How are ag and ag2 different?!
     @staticmethod
     @pytest.fixture()
     def ag2():
         u2 = MDAnalysis.Universe(GRO)
-        return u2.atoms[:20]
+        return u2.atoms[11:20]
 
+    @staticmethod
+    @pytest.fixture()
+    def box():
+        return np.array([8, 8, 8, 90, 90, 90], dtype=np.float32)
+    
     @staticmethod
     @pytest.fixture()
     def expected(ag, ag2):
-        np.random.shuffle(ag2.positions)
+        
         return np.diag(scipy.spatial.distance.cdist(
             ag.positions, ag2.positions)
         )
 
+    @staticmethod
+    @pytest.fixture()
+    def expected_box(ag, ag2, box):
+
+        rp = np.abs(ag.positions - ag2.positions)
+        for i in range(len(rp)):
+            for j in range(len(rp[i])):
+                rp[i][j] = np.where(rp[i][j] > box[j]/2, box[j] - rp[i][j], rp[i][j])
```

Comment:
`np.where` doesn't make sense with scalars. Broadcast `box` to two dimensions and get rid of the double loop

```
box_2d = box[np.newaxis, 0:3]
rp = np.where(rp  > box_2d / 2, box_2d - rp, rp)
```